qubesmanager/bootfromdevice.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/python2
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#
+
+import subprocess
+from . import utils
+from .firewall import *
+from .ui_bootfromdevice import *
+import qubesadmin.tools.qvm_start as qvm_start
+
+
+class VMBootFromDeviceWindow(Ui_BootDialog, QDialog):
+    def __init__(self, vm, qapp, parent=None):
+        super(VMBootFromDeviceWindow, self).__init__(parent)
+
+        self.vm = vm
+        self.qapp = qapp
+
+        self.setupUi(self)
+        self.setWindowTitle(self.tr("Boot {vm} from device").format(vm=self.vm.name))
+
+        self.connect(self.buttonBox, SIGNAL("accepted()"), self.save_and_apply)
+        self.connect(self.buttonBox, SIGNAL("rejected()"), self.reject)
+
+        # populate buttons and such
+        self.__init_buttons__()
+
+
+    def reject(self):
+        self.done(0)
+
+    def save_and_apply(self):
+        if self.blockDeviceRadioButton.isChecked():
+            full_path = self.blockDeviceComboBox.currentText()
```

Comment:
`full_path` suggests it is a path, while in fact it is device identifier. Or a path prefixed with VM name. Better would be `cdrom_location` or sth lilke this.